File: Source/Hafner.Compatibility.MetaPackage.CompileTests.CS/Hafner.Compatibility.CallerInformationAttributes/TestClass.cs
namespace Hafner.Compatibility.CompileTest.CS.CallerInformationAttributes;

using System.Runtime.CompilerServices;

public static class TestClass {

    public static string GetCallerInfoG1([CallerFilePath] string filePath = "",
                               [CallerLineNumber] int lineNumber = 0,
                               [CallerMemberName] string memberName = "") {
        return $"File: {filePath}, Line: {lineNumber}, Member: {memberName}";
    }

    public static string GetCallerInfoG2(string arg1, int arg2, [CallerArgumentExpression(nameof(arg1))] string arg1Name = "", [CallerArgumentExpression(nameof(arg2))] string arg2Name = "") {
        return $"Name/expression of arg1: {arg1Name}, Name/expression of arg2: {arg2Name}";
    }

}

File: Source/Hafner.Compatibility.MetaPackage.CompileTests.VB/Hafner.Compatibility.CallerInformationAttributes/TestClass.vb
Imports System.Runtime.CompilerServices

Namespace CallerInformationAttributesTests

    Public Class TestClass

        Public Shared Function GetCallerInfoG1(<CallerFilePath> Optional filePath As String = "",
                                               <CallerLineNumber> Optional lineNumber As Int32 = 0,
                                               <CallerMemberName> Optional memberName As String = "") As String

            Return $"File: {filePath}, Line: {lineNumber}, Member: {memberName}"
        End Function

        Public Shared Function GetCallerInfoG2(arg1 As String,
                                               arg2 As Int32,
                                               <CallerArgumentExpression("arg1")> arg1Name As String,
                                               <CallerArgumentExpression("arg2")> arg2Name As String) As String

            Return $"Name/expression of arg1: {arg1Name}, Name/expression of arg2: {arg2Name}"
        End Function

    End Class

End Namespace
